lib/lib-table-query/src/test/java/org/sagebionetworks/table/query/SqlElementUntilsTest.java

```diff
@@ -82,4 +76,148 @@ public void testReplaceSorted() throws ParseException {
+		assertEquals("SELECT * FROM syn123 LIMIT 34 OFFSET 12", converted.toString());
+	}
+	
+	@Test
+	public void testOverridePaginationQueryWithLimitOffsetOverrideLimitNull() throws ParseException{
+		QuerySpecification model = TableQueryParser.parserQuery("select * from syn123 limit 34 offset 12");
+		Long limit = null;
+		Long offset = 2L;
+		Long maxRowsPerPage = 1000L;
+		QuerySpecification converted = SqlElementUntils.overridePagination(model, offset, limit, maxRowsPerPage);
+		assertEquals("SELECT * FROM syn123 LIMIT 32 OFFSET 14", converted.toString());
+	}
+	
+	@Test
+	public void testOverridePaginationQueryWithLimitOffsetOverrideOffsetNull() throws ParseException{
+		QuerySpecification model = TableQueryParser.parserQuery("select * from syn123 limit 34 offset 12");
+		Long limit = 3L;
+		Long offset = null;
+		Long maxRowsPerPage = 1000L;
+		QuerySpecification converted = SqlElementUntils.overridePagination(model, offset, limit, maxRowsPerPage);
+		assertEquals("SELECT * FROM syn123 LIMIT 3 OFFSET 12", converted.toString());
+	}
+	
+	@Test
+	public void testOverridePaginationQueryWithLimitOffsetOverrideInRange() throws ParseException{
+		QuerySpecification model = TableQueryParser.parserQuery("select * from syn123 limit 100 offset 50");
+		Long limit = 25L;
+		Long offset = 10L;
+		Long maxRowsPerPage = 1000L;
+		QuerySpecification converted = SqlElementUntils.overridePagination(model, offset, limit, maxRowsPerPage);
+		assertEquals("SELECT * FROM syn123 LIMIT 25 OFFSET 60", converted.toString());
+	}
+	
+	@Test
+	public void testOverridePaginationQueryWithLimitOffsetOverrideOutOfRange() throws ParseException{
+		QuerySpecification model = TableQueryParser.parserQuery("select * from syn123 limit 100 offset 75");
+		Long limit = 50L;
+		Long offset = 10L;
+		Long maxRowsPerPage = 1000L;
+		QuerySpecification converted = SqlElementUntils.overridePagination(model, offset, limit, maxRowsPerPage);
+		assertEquals("SELECT * FROM syn123 LIMIT 50 OFFSET 85", converted.toString());
+	}
+	
+	@Test
+	public void testOverridePaginationMaxRowPerPageNull() throws ParseException{
+		QuerySpecification model = TableQueryParser.parserQuery("select * from syn123 limit 100 offset 75");
+		Long limit = 50L;
+		Long offset = 10L;
+		Long maxRowsPerPage = null;
+		QuerySpecification converted = SqlElementUntils.overridePagination(model, offset, limit, maxRowsPerPage);
+		assertEquals("SELECT * FROM syn123 LIMIT 50 OFFSET 85", converted.toString());
+	}
+	
+	@Test
+	public void testOverridePaginationAllNull() throws ParseException{
+		QuerySpecification model = TableQueryParser.parserQuery("select * from syn123 limit 100 offset 75");
+		Long limit = null;
+		Long offset = null;
+		Long maxRowsPerPage = null;
+		QuerySpecification converted = SqlElementUntils.overridePagination(model, offset, limit, maxRowsPerPage);
+		assertEquals("SELECT * FROM syn123 LIMIT 100 OFFSET 75", converted.toString());
+	}
+	

```

Comment:
Need two new tests: maxRowsPerPage < override but > query limit, maxRowsPerPage < override and < query limit.